net/tcp/tcp_conn.c

```diff
@@ -699,9 +717,10 @@ FAR struct tcp_conn_s *tcp_alloc(uint8_t domain)
 
   /* Allocate the connect entry from heap */
 
-#ifdef CONFIG_NET_ALLOC_CONNS
+#if CONFIG_NET_TCP_ALLOC_CONNS > 0
   if (conn == NULL)
     {
+      alloc_conn = 1;
```

Comment:
the allocation check is wrong since the free connection got above may dynamically allocate too, but put to the free list in tcp_free.

Reply:
Done.